### ISSUE TITLE
T, P, RH% sensor for hardware

### ISSUE THREAD

**Issue by PaoloGB** [Rymapt/ethoscope#21]

**Comment by PaoloGB** [comment]
A few options are available.
For instance [Honeywell](http://uk.rs-online.com/web/p/temperature-humidity-sensors/7811368/)
has a temperature/humidity sensor we could use. Inerface over I2C.
At this stage the main limitation we have is that we can only use components we can solder by hand (until we go for a PCB shield production but that is probably ~30/ board and a minimum of 10) so a [breakout pcb](http://uk.farnell.com/roth-elektronik/re899/ic-adaptor-fibreglass-soic-sop/dp/2474677?ost=2474677&searchView=table&iscrfnonsku=false&ddkey=http%3Aen-GB%2FElement14_United_Kingdom%2Fsearch) is also necessary.
So, shopping list:

-sensor RS 781-1368 (£6.63/chip)
-pcb breakout Farnell 2474677 (£1.54/each, minimum 10)
-2.2kohm
-0.22 uF


[Documentation](https://learn.sparkfun.com/tutorials/raspberry-pi-spi-and-i2c-tutorial).

**Comment by PaoloGB** [comment]
- The sensor is working and I have an (ugly) python script to poll it and read Temperature and Humidity.
- Some calibration might be required. Maybe a test with the environmental chamber could be useful. I suspect some calibration parameters might be necessary in the code.
- The script is ugly because it talks to the I2C bus but not in the way the sensor is expecting. The sensor seems to require a method not implemented in the SMBUS library, which is what everyone seems to use to talk to I2C with the Pi.
- It might be possible to write our own method to perform the read according the the manufacturer's specifications. I might ask you guys for some help for this step, as I could not figure out where the methods are defined.
- Bottom line is that, if needed, this could be used already.

**Comment by PaoloGB** [comment]
I now have a script that performs the I2C operations correctly. It uses two different libraries for I2C:
SMBus for writing
[easyi2c](http://www.pichips.co.uk/index.php/RPi_I2C_2017) for reading.
This should not be an issue.
The sensor readings seem to make sense.

There is an option to put the [HIH6130](https://sensing.honeywell.com/HIH6130-021-001-humidity-sensors) in [command mode](https://sensing.honeywell.com/command-mode-humidicon-tn-009062-3-en-final-07jun12.pdf) where it is possible to write to the EEPROM and change the I2C address and set threshold alarms for the humidity. Entering command mode is tricky: you have to issue a command within 10 ms from powering up the device, so for now I am leaving those options out.

**TO DO**: define how we want the function to be accessible, i.e. do we want a class? Just a function? How to we poll from the main code?
